lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_version_helper.rb

```diff
@@ -127,81 +265,162 @@ def self.bump_version_release
+        # @param [String] keyword The keyword (key name) we want the value for
+        #
+        # @return [String] Returns the value for that keyword in the section of the `.gradle` file, or nil if not found.
+        #
+        def self.get_keyword_from_gradle_file(file_path, section, keyword)
           found_section = false
           File.open(file_path, 'r') do |file|
             file.each_line do |line|
               if !found_section
-                if (line.include? section) 
+                if line.include?(section)
                   found_section = true
                 end
               else
-                if (line.include? keyword)
-                  return line unless line.include?("\"#{keyword}\"") or line.include?("P#{keyword}")
+                if line.include?(keyword) && !line.include?("\"#{keyword}\"") && !line.include?("P#{keyword}")
+                  return line.split(' ')[1]
                 end
               end
             end
           end
           return nil
         end
 
+        # Ensure that a version string is correctly formatted (that is, each of its parts is a number) and returns the 2-parts version number
+        #
+        # @param [String] version The version string to verify
+        #
+        # @return [String] The "major.minor" version string, only with the first 2 components
+        # @raise [UserError] If any of the parts of the version string is not a number
+        #
         def self.verify_version(version)
           v_parts = get_version_parts(version)
           
           v_parts.each do | part |
-            if (!is_number?(part)) then
+            if (!is_int?(part)) then
               UI.user_error!("Version value can only contains numbers.")
             end
           end
 
           "#{v_parts[MAJOR_NUMBER]}.#{v_parts[MINOR_NUMBER]}"
         end
 
-        def self.is_number? string
-          true if Float(string) rescue false
+        # Check if a string is an integer.
+        #
+        # @param [String] string The string to test
+        #
+        # @return [Bool] true if the string is representing an integer value, false if not
+        #
+        def self.is_int? string
+          true if Integer(string) rescue false
         end
 
+        # The path to the build.gradle file for the project.
+        #
+        # @env PROJECT_ROOT_FOLDER The path to the root of the project (the folder containing the `.git` directory).
+        # @env PROJECT_NAME The name of the project, i.e. the name of the subdirectory containing the project's `build.gradle` file.
+        #
+        # @return [String] The path of the `build.gradle` file inside the project subfolder in the project's repo
+        #
         def self.gradle_path 
-          "#{ENV["PROJECT_ROOT_FOLDER"]}#{ENV["PROJECT_NAME"]}/build.gradle"
+          File.join(ENV["PROJECT_ROOT_FOLDER"] || '.', ENV["PROJECT_NAME"], 'build.gradle')
```

Comment:
If I'm not mistaken, wouldn't this would presume that the root `build.gradle` is in the `fastlane` directory if `ENV["PROJECT_ROOT_FOLDER"]` isn't set?

I'm not sure that matches the setup of our repos.

IMHO it would make sense to make this a failing error – it could explode with a helpful error message if `ENV["PROJECT_ROOT_FOLDER"] ` isn't available?

WDYT?

Reply:
Good point and good idea

Reply:
Addressed in 0813fe61d73fbbf2263a961a7d85a027a4ac4dac